File: sdks/csharp/src/NovaBilling/Types/AssignTaxDto.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record AssignTaxDto : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// Tax ID to assign
    /// </summary>
    [JsonPropertyName("taxId")]
    public required string TaxId { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Types/AssignTaxDto.php
<?php

namespace NovaBilling\Types;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class AssignTaxDto extends JsonSerializableType
{
    /**
     * @var string $taxId Tax ID to assign
     */
    #[JsonProperty('taxId')]
    public string $taxId;

    /**
     * @param array{
     *   taxId: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->taxId = $values['taxId'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
